pkg/pillar/evetpm/decrypt.go

```diff
@@ -68,12 +69,25 @@ func getDecryptKey(log *base.LogObject, X, Y *big.Int, edgeNodeCert *types.EdgeN
 	return decryptKey, nil
 }
 
-func aesDecrypt(log *base.LogObject, plaintext, ciphertext, key, iv []byte) error {
-	aesBlockDecrypter, err := aes.NewCipher([]byte(key))
+//AESEncrypt encrypts plaintext, and returns it in ciphertext
+//by using the key and initial value given. Uses a AES CFB cipher.
+func AESEncrypt(ciphertext, plaintext, key, iv []byte) error {
+	aesBlockEncrypter, err := aes.NewCipher([]byte(key))
 	if err != nil {
-		log.Errorf("creating aes new cipher failed: %v", err)
 		return err
 	}
+	aesEncrypter := cipher.NewCFBEncrypter(aesBlockEncrypter, iv)
+	aesEncrypter.XORKeyStream(ciphertext, plaintext)
+	return nil
+}
+
+//AESDecrypt decrypts ciphertext, and returns it in plaintext
+//using the key and initial value given. Uses AES CFB cipher.
+func AESDecrypt(plaintext, ciphertext, key, iv []byte) error {
```

Comment:
If I have the wrong key, will this return an error? Or will it return data (which is nonsensical)?
Thus do we need a hmac or sha over the data to verify it is correct post decrypt?

Reply:
Added a field in protobuf to carry SHA256 digest of the key, for validating decrypted output